momento-sdk/src/main/java/momento/sdk/TopicClient.java

```diff
@@ -0,0 +1,104 @@
+package momento.sdk;
+
+import java.io.Closeable;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+import momento.sdk.auth.CredentialProvider;
+import momento.sdk.config.Configuration;
+import momento.sdk.responses.topic.TopicPublishResponse;
+import momento.sdk.responses.topic.TopicSubscribeResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Client to perform operations against the Momento Topic Service */
+public final class TopicClient implements Closeable {
+  private final ScsTopicClient scsTopicClient;
+
+  private final Logger logger = LoggerFactory.getLogger(TopicClient.class);
+
+  /**
+   * Constructs a TopicClient.
+   *
+   * @param credentialProvider Provider for the credentials required to connect to Momento.
+   * @param configuration Configuration object containing all tunable client settings.
+   */
+  public TopicClient(
+      @Nonnull CredentialProvider credentialProvider, @Nonnull Configuration configuration) {
+    this.scsTopicClient = new ScsTopicClient(credentialProvider, configuration);
+    logger.info("Creating Momento Topic Client");
+    logger.debug("Cache endpoint: " + credentialProvider.getCacheEndpoint());
+    logger.debug("Control endpoint: " + credentialProvider.getControlEndpoint());
+  }
+
+  /**
+   * Constructs a TopicClient.
+   *
+   * @param credentialProvider Provider for the credentials required to connect to Momento.
+   * @param configuration Configuration object containing all tunable client settings.
+   * @return TopicClient
+   */
+  public static TopicClient create(
+      @Nonnull CredentialProvider credentialProvider, @Nonnull Configuration configuration) {
+    return create(credentialProvider, configuration);
+  }
+
+  /**
+   * Creates a TopicClient builder.
+   *
+   * @param credentialProvider Provider for the credentials required to connect to Momento.
+   * @param configuration Configuration object containing all tunable client settings.
+   * @return The builder.
+   */
+  public static TopicClientBuilder builder(
+      CredentialProvider credentialProvider, Configuration configuration) {
+    return new TopicClientBuilder(credentialProvider, configuration);
+  }
+
+  /**
+   * Publish a message to a topic with provided topic name in a cache with provided cache name.
+   *
+   * @param cacheName The name of the cache where topic resides.
+   * @param topicName The name of the topic.
+   * @param message The message to be published.
+   * @return A future containing the result of the topic publish: {@link
+   *     TopicPublishResponse.Success} or {@link TopicPublishResponse.Error}.
+   */
+  public CompletableFuture<TopicPublishResponse> publish(
```

Comment:
when playing with error handling on these I noticed that if I specify a cache that does not exist, it seems that we throw the NOT_FOUND exception (as opposed to returning an `Error` response).

I am okay with this as long as:
(a) this is how we are handling it elsewhere in the Java SDK (e.g. cache set/get), and
(b) we are returning the Error response type for other errors? otherwise it doesn't make any sense to even have the error response type if we are just always going to throw.

can you verify this?

Reply:
I modified my program to run a bunch of threads that all do Publishes in a loop, to intentionally exceed the throttling limit, and AFAICT the first time I hit the throttling limit an exception is thrown (rather than an Error response being returned).

This seems inconsistent with other SDKs as far as I know? Consistency is my top priority so if I'm wrong about what happens in other SDKs we can discuss.

But unless i'm doing something silly, it seems like the `Error` types in this PR may not be getting used at all yet.